hydra-node/src/Hydra/Chain/Backend.hs

```diff
@@ -0,0 +1,38 @@
+module Hydra.Chain.Backend where
+
+import Hydra.Prelude
+
+import Hydra.Cardano.Api (
+  Address,
+  ChainPoint,
+  EraHistory,
+  GenesisParameters,
+  LedgerEra,
+  NetworkId,
+  PParams,
+  PaymentKey,
+  PoolId,
+  ShelleyAddr,
+  ShelleyEra,
+  SystemStart (..),
+  Tx,
+  TxId,
+  UTxO,
+  VerificationKey,
+ )
+import Hydra.Chain.CardanoClient qualified as CardanoClient
+import Hydra.Tx (ScriptRegistry)
+
+class ChainBackend a where
```

Comment:
in love with this type class! <3

Reply:
I'm not a fan of this kind of lawless overloading, but if we must do this then it should be a monad class. methods should not have constraints in the context.

An example of this kind of style is in pandoc:

https://hackage-content.haskell.org/package/pandoc-3.7.0.1/docs/Text-Pandoc-Class.html#t:PandocMonad

Reply:
@locallycompact I lean towards your approach but even more I think handles would be appropriate considering the our coding style. Let's do the changes as a follow up if you agree since this PR is dragging for a long time.